### PR TITLE
GUI-2647 ELB Wizard, begin rewrite, shell

### PR DESCRIPTION
**Pull request by coyote240**
https://eucalyptus.atlassian.net/browse/GUI-2647

- Add new endpoint to allow new wizard to be developed in parallel with existing wizard
- Introduce ngRoute
- Page router and step controllers for new wizard
- Begin listener-editor, completed in next PR for GUI-2751
- Tests

### CODE REVIEW

File: eucaconsole/static/css/thirdparty/font-awesome.css [hunk 1]
```diff
@@ -6,8 +6,8 @@
  * -------------------------- */
 @font-face {
   font-family: 'FontAwesome';
-  src: url('../fonts/font-awesome/fontawesome-webfont.eot?v=4.3.0');
-  src: url('../fonts/font-awesome/fontawesome-webfont.eot?#iefix&v=4.3.0') format('embedded-opentype'), url('../fonts/font-awesome/fontawesome-webfont.woff2?v=4.3.0') format('woff2'), url('../fonts/font-awesome/fontawesome-webfont.woff?v=4.3.0') format('woff'), url('../fonts/font-awesome/fontawesome-webfont.ttf?v=4.3.0') format('truetype'), url('../fonts/font-awesome/fontawesome-webfont.svg?v=4.3.0#fontawesomeregular') format('svg');
+  src: url('../fonts/fontawesome-webfont.eot?v=4.3.0');
```
Comment (dkavanagh) [inline eucaconsole/static/css/thirdparty/font-awesome.css]: I *think* we must modify the path here due to us not storing it where FA thinks we are.

**Comment by dkavanagh** [comment]
should eucaRoutes service be used for most (if not all) of the paths currently hard coded?

**Comment by coyote240** [comment]
None of the paths hard coded are routes.  All of these are paths entirely defined within the ngRoute, and only exist on this page.

File: eucaconsole/static/js/thirdparty/nvd3/nv.d3.js [hunk 2]
```diff
@@ -4881,9 +4881,14 @@ nv.models.lineChart = function() {
...
 
+                if ( margin.top != legend.height()) {
+                    margin.top = legend.height();
+                    availableHeight = (height || parseInt(container.style('height')) || 400)
+                        - margin.top - margin.bottom;
+                }
+
                 wrap.select('.nv-legendWrap')
-                    //.attr('transform', 'translate(0,' + (-margin.top) +')')
-                    .attr('transform', 'translate(10, 240)')
```
Comment (dkavanagh) [inline eucaconsole/static/js/thirdparty/nvd3/nv.d3.js]: The above was a hand-tweak to relocate legends.

**Comment by dkavanagh** [comment]
Not sure if this should be part of this PR, but we've shown the next button "disabled" till the current  tab is valid and it looks "enabled" here. Same with tab nav. We've disallowed forward nav if current tab is not valid, but back and forth on validated tabs OK.

**Comment by coyote240** [comment]
@dkavanagh This will have to come in a further PR.  I've created GUI-2763 to address this feature.

File: eucaconsole/static/js/thirdparty/nvd3/nv.d3.js (hunk 2 above)
Comment (coyote240) [inline eucaconsole/static/js/thirdparty/nvd3/nv.d3.js]: This is getting problematic.  Let's work together to address either a set of patches or some other method of overrides for third party libraries.  Or, in the bower config, lock the version outright.

File: eucaconsole/static/css/thirdparty/font-awesome.css (hunk 1 above)
Comment (coyote240) [inline eucaconsole/static/css/thirdparty/font-awesome.css]: Again, keeping track of hand-coded tweaks to third-party libraries needs to be managed.  We can use the bower config and package.json to fix versions if necessary.

**Pull request merged**